Add support for image digests in helm charts

aanm opened the pull request:
Read per commit

aanm committed:
contrib/release: do not require images to be download locally

The SHAs are available in the GH run, so we can use the GH API to
retrieve them automatically.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: generate release files automatically

Having files with the right format will make it easier to download them
locally and use them as part of the release process.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
install: add digests into helm charts

This commit adds a simple way to add the image digests into the official
helm charts.

Signed-off-by: André Martins <andre@cilium.io>

sayboras commented on install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml:
```diff
@@ -193,7 +193,11 @@ spec:
 {{- with .Values.extraEnv }}
 {{ toYaml . | trim | indent 8 }}
 {{- end }}
+{{- if .Values.image.useDigest }}
```
any reason not to use .Values.image.digest here directly (i.e. check if .Values.image.digest is not empty) ?

aanm replied:
@sayboras see https://github.com/cilium/cilium/pull/15185#discussion_r586700040 if it makes sense

joestringer commented on .github/workflows/images-legacy-releases.yaml:
```diff
@@ -16,61 +16,80 @@ jobs:
+
       - name: Release Build ${{ matrix.name }}
-        uses: docker/build-push-action@4a531fa5a603bab87dfa56578bd82b28508c9547 # v2.2.2
+        uses: docker/build-push-action@9379083e426e2e84abb80c8c091f5cdeb7d3fd7a # v2.2.2
```
Is this still `v2.2.2`? The sha changed.

aanm replied:
Not it is not, I need to remove all comments since dependabot will update those SHAs automatically.

joestringer commented on install/kubernetes/Makefile:
```diff
@@ -56,14 +56,35 @@ LOGO_PATH := Documentation/images/logo-solo.svg
-	$(QUIET)helm template cilium cilium --namespace=kube-system $(QUICK_HUBBLE_OPTIONS) > $(QUICK_HUBBLE_INSTALL)
-
+	$(QUIET)helm template cilium cilium --namespace=kube-system \
+     --set image.useDigest=false \
+     --set operator.image.useDigest=false \
+     --set preflight.image.useDigest=false \
+     --set hubble.relay.image.useDigest=false \
+     --set clustermesh.apiserver.image.useDigest=false \
```
Presumably these new parameters should be no-ops because the default is false.

aanm replied:
Yes, once we release 1.9, we need to change them to true. (the same does not apply to 1.7 as we already release images based on GH actions)

aanm committed:
.github: remove comments of GH action versions

Dependabot will update those commit SHAs automatically and the version
number will quickly outdated. To avoid confusion, we will remove the
comments related with the versions a SHA points to from all GH actions.

Signed-off-by: André Martins <andre@cilium.io>
